### PR TITLE
[CoSim] Adding interface for CoSimIO V1.0.0

### PR DESCRIPTION
**Pull request by philbucher**
This adds the interface for the [CoSimIO V1.0.0](https://github.com/KratosMultiphysics/CoSimIO/releases/tag/v1.0.0)

Regarding the mesh-exchange we should thing what to do if we want to communicate 
- only nodes
- Conditions
- Current vs Initial configuration

### CODE REVIEW

**Comment by philbucher** [comment]
After this is in the final step for using it will be to add the corresponding python classes for `SolverWrapper` and `CoSimulationIO`

File: applications/CoSimulationApplication/tests/co_sim_io_py_exposure_aux_files/connect_disconnect.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+from KratosMultiphysics.CoSimulationApplication import CoSimIO
+
+connection_settings = CoSimIO.Info()
+connection_settings.SetString("connection_name", "c_d_test")
+connection_settings.SetInt("echo_level", 0)
+CoSimIO.Connect(connection_settings)
+
```
Comment (pooyan-dadvand) [inline applications/CoSimulationApplication/tests/co_sim_io_py_exposure_aux_files/connect_disconnect.py:9]: Checking the status of the connection would be useful here

File: applications/CoSimulationApplication/custom_python/add_co_sim_io_to_python.cpp [hunk 2]
```diff
@@ -0,0 +1,481 @@
...
+    // // m_co_sim_io.def("ImportGeometry", CoSimIO_Wrappers::ImportGeometry); // This is not yet implemented in the CoSimIO
+    // // m_co_sim_io.def("ExportGeometry", CoSimIO_Wrappers::ExportGeometry); // This is not yet implemented in the CoSimIO
+
+    py::enum_<CoSimIO_Wrappers::DataLocation>(m_co_sim_io,"DataLocation")
+        .value("NodeHistorical",    CoSimIO_Wrappers::DataLocation::NodeHistorical)
+        .value("NodeNonHistorical", CoSimIO_Wrappers::DataLocation::NodeNonHistorical)
+        .value("Element",           CoSimIO_Wrappers::DataLocation::Element)
+        .value("Condition",         CoSimIO_Wrappers::DataLocation::Condition)
+        .value("ModelPart",         CoSimIO_Wrappers::DataLocation::ModelPart)
+        ;
```
Comment (pooyan-dadvand) [inline applications/CoSimulationApplication/custom_python/add_co_sim_io_to_python.cpp:459]: ProcessInfo?

Comment (pooyan-dadvand) [reply applications/CoSimulationApplication/custom_python/add_co_sim_io_to_python.cpp:459]: Element/Condition gauss points values?

**Review by pooyan-dadvand — COMMENTED** [review]
Nice work! Some minor comments

File: applications/CoSimulationApplication/custom_python/add_co_sim_io_to_python.cpp (hunk 2 above)
Comment (philbucher) [reply applications/CoSimulationApplication/custom_python/add_co_sim_io_to_python.cpp:459]: will be added in the future

File: applications/CoSimulationApplication/tests/co_sim_io_py_exposure_aux_files/connect_disconnect.py (hunk 1 above)
Comment (philbucher) [reply applications/CoSimulationApplication/tests/co_sim_io_py_exposure_aux_files/connect_disconnect.py:9]: done!

**Comment by philbucher** [comment]
@pooyan-dadvand I added the checks for the `ConnectionStatus`

Now it is ready

**Review by pooyan-dadvand — APPROVED** [review]
Thanks!

**Pull request merged**